src/dense/generic_dense.jl

```diff
@@ -738,62 +778,57 @@ end
     #out = similar(y₀,axes(idxs))
     #@views @.. broadcast=false out = (-4*dt*k[1][idxs] - 2*dt*k[2][idxs] - 6*y₀[idxs] + Θ*(6*dt*k[1][idxs] + 6*dt*k[2][idxs] + 12*y₀[idxs] - 12*y₁[idxs]) + 6*y₁[idxs])/(dt*dt)
-    @views out = (-4 * dt * k[1][idxs] - 2 * dt * k[2][idxs] - 6 * y₀[idxs] +
+    @views out = dv[idxs] .* (-4 * dt * k[1][idxs] - 2 * dt * k[2][idxs] - 6 * y₀[idxs] +
                   Θ * (6 * dt * k[1][idxs] + 6 * dt * k[2][idxs] + 12 * y₀[idxs] -
                    12 * y₁[idxs]) + 6 * y₁[idxs]) / (dt * dt)
     out
 end
 
-@muladd function hermite_interpolant!(out, Θ, dt, y₀, y₁, k, idxs::Nothing, T::Type{Val{2}}) # Default interpolant is Hermite
-    @inbounds @.. broadcast=false out=(-4 * dt * k[1] - 2 * dt * k[2] - 6 * y₀ +
+@muladd function hermite_interpolant!(out, Θ, dt, y₀, y₁, k, idxs::Nothing, T::Type{Val{2}}, dv=true) # Default interpolant is Hermite
+    @inbounds @.. broadcast=false out= dv * (-4 * dt * k[1] - 2 * dt * k[2] - 6 * y₀ +
                                        Θ *
                                        (6 * dt * k[1] + 6 * dt * k[2] + 12 * y₀ - 12 * y₁) +
                                        6 * y₁) / (dt * dt)
 end
 
 @muladd function hermite_interpolant!(out::Array, Θ, dt, y₀, y₁, k, idxs::Nothing,
-    T::Type{Val{2}}) # Default interpolant is Hermite
+    T::Type{Val{2}}, dv=trues(size(out))) # Default interpolant is Hermite
     @inbounds @simd ivdep for i in eachindex(out)
-        out[i] = (-4 * dt * k[1][i] - 2 * dt * k[2][i] - 6 * y₀[i] +
+        out[i] = dv[i] * (-4 * dt * k[1][i] - 2 * dt * k[2][i] - 6 * y₀[i] +
                   Θ * (6 * dt * k[1][i] + 6 * dt * k[2][i] + 12 * y₀[i] - 12 * y₁[i]) +
                   6 * y₁[i]) / (dt * dt)
     end
     out
 end
 
-@muladd function hermite_interpolant!(out, Θ, dt, y₀, y₁, k, idxs, T::Type{Val{2}}) # Default interpolant is Hermite
-    @views @.. broadcast=false out=(-4 * dt * k[1][idxs] - 2 * dt * k[2][idxs] -
+@muladd function hermite_interpolant!(out, Θ, dt, y₀, y₁, k, idxs, T::Type{Val{2}}, dv=true) # Default interpolant is Hermite
+    @views @.. broadcast=false out=dv * (-4 * dt * k[1][idxs] - 2 * dt * k[2][idxs] -
                                     6 * y₀[idxs] +
                                     Θ * (6 * dt * k[1][idxs] + 6 * dt * k[2][idxs] +
                                      12 * y₀[idxs] - 12 * y₁[idxs]) + 6 * y₁[idxs]) /
                                    (dt * dt)
 end
 
-@muladd function hermite_interpolant!(out::Array, Θ, dt, y₀, y₁, k, idxs, T::Type{Val{2}}) # Default interpolant is Hermite
-    @views @.. broadcast=false out=(-4 * dt * k[1][idxs] - 2 * dt * k[2][idxs] -
-                                    6 * y₀[idxs] +
-                                    Θ * (6 * dt * k[1][idxs] + 6 * dt * k[2][idxs] +
-                                     12 * y₀[idxs] - 12 * y₁[idxs]) + 6 * y₁[idxs]) /
-                                   (dt * dt)
```

Comment:
this one had two complete copies of the interpolation??? I just kept the last one which was effectively being used I guess.